lib/lz4hc.c

```diff
@@ -353,7 +434,7 @@ static int LZ4HC_compress_hashChain (
     /* Main Loop */
     while (ip < mflimit) {
         ml = LZ4HC_InsertAndFindBestMatch (ctx, ip, matchlimit, (&ref), maxNbAttempts);
```

Comment:
I think we need to pass `start0` to this call to be updated.

Reply:
Not sure to follow this one.
`start0` is updated _after_ a successful call to `LZ4HC_InsertAndFindBestMatch()`, in which case it is set to `ip`.
It does not seem useful to update `start0` before, as the function is just looping over `LZ4HC_InsertAndFindBestMatch()` until it finds at least one match.

Reply:
I was incorrect because `LZ4HC_InsertAndFindBestMatch()` won't update `startpos`.